tests/unit/widget/widget_classes.js

```diff
@@ -0,0 +1,131 @@
+		}
+	}
+});
+
+$.widget( "ui.classesWidget", {
+	options: {
+		classes: {
+			"test-wrapper": "self-wrapper",
+			"test-self": "self-class self-class-2"
+		}
+	},
+	_create: function() {
+		this.span = $( "<span>" )
+			.appendTo( this.element );
+
+		this.element.wrap( "<div>" );
+
+		this.wrapper = this.element.parent();
+
+		this._addClass( "test-self", "test-self-extra" )
+			._addClass( "test-self-2" )
+			._addClass( null, "test-self-extra-null" )
+			._addClass( this.span, null, "test-span-extra-null" )
+			._addClass( this.span, "test-span", "test-span-extra" )
+			._addClass( this.wrapper, "test-wrapper" );
+
+	},
+	toggleClasses: function( bool ) {
+		this._toggleClass( "test-self", "test-self-extra", bool )
+			._toggleClass( "test-self-2", null, bool )
+			._toggleClass( null, "test-self-extra-null", bool )
+			._toggleClass( this.span, null, "test-span-extra-null", bool )
+			._toggleClass( this.span, "test-span", "test-span-extra", bool )
+			._toggleClass( this.wrapper, "test-wrapper", null, bool );
+	},
+	removeClasses: function() {
+		this._removeClass( "test-self", "test-self-extra" )
+			._removeClass( "test-self-2" )
+			._removeClass( null, "test-self-extra-null" )
+			._removeClass( this.span, null, "test-span-extra-null" )
+			._removeClass( this.span, "test-span", "test-span-extra" )
+			._removeClass( this.wrapper, "test-wrapper" );
+	},
+	_destroy: function() {
+		this.span.remove();
+		this.element.unwrap();
+	},
+	widget: function() {
+		return this.wrapper;
+	}
+});
+
+test( ".option() - classes setter", function( assert ){
+	expect( 12 );
+
+	var testWidget = $.ui.classesWidget(),
+		currentWrapperClass = testWidget.option( "classes.test-wrapper" );
+
+	testElementClasses( testWidget.element, true, "add" );
+
+	testWidget.option({
+		classes: {
+			"test-span": "self-span-new",
+			"test-wrapper": currentWrapperClass + " self-wrapper-new",
+			"test-self": "self-class-2"
+		}
+	});
+
+	assert.hasClasses( testWidget.element, "test-self self-class-2",
+		"Removing a class leaves the structure and other classes in value" );
+	assert.lacksClasses( testWidget.element, "self-class",
+		"Removing a class from the value removes the class" );
+	testWidget.option( "classes.test-self", "" );
+	assert.hasClasses( testWidget.element, "test-self",
+		"Setting to empty value leaves structure class" );
+	assert.lacksClasses( testWidget.element, "self-class-2",
+		"Setting empty value removes previous value classes" );
+	assert.hasClasses( testWidget.span, "test-span self-span-new",
+		"Adding a class to an empty value works as expected" );
+	assert.hasClasses( testWidget.wrapper, "test-wrapper self-wrapper-new",
+		"Appending a class to the current value works as expected" );
+});
+
+test( ".destroy() - class removal", function(){
+	expect( 1 );
+
+	domEqual( "#widget", function(){
+		$( "#widget" ).classesWidget().classesWidget( "destroy" );
+	});
+});
+
+function testElementClasses( widget, bool, method ) {
```

Comment:
This is too clever. I suggest splitting this up into two helpers, both using the same number of assertions, but hardcoding everything, removing the need for the `bool` argument. Also moving it into the scope of the only test that uses it, which would also allow to use the `assert` argument properly and remove the `widget` argument.